feat: add nanminmax package to handle NaN values

abhishekblue opened the pull request:
Resolves #5579 

## Description

> What is the purpose of this pull request?

This pull request:

-   introduces the @stdlib/stats/incr/nanminmax package, which incrementally computes minimum and maximum, ignoring NaN values.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #5579 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

abhishekblue committed:
feat: add nanminmax package to handle NaN values

abhishekblue committed:
update main.js

Neerajpathak07 commented:
@abhishekblue So, for every example the return value is:- `[ -5, 5 ]` even if the input values are different.
This would be primarily because the `main.js` implementation isn't in-line I'd recommend you to go through it again.
you can also run `eslint` locally to check for any errors.

abhishekblue commented:
@Neerajpathak07 I have checked tests again, also checked by logging the values of the examples it works fine, also checked modules as well. Any other suggestions?

Neerajpathak07 commented:
@abhishekblue Ig you did everything except giving the `main.js` file a brief look. since we are testing the values for `accumulator` function and it is coming inconsistent then definitely the function isn't correct.

kgryte commented on lib/node_modules/@stdlib/stats/incr/nanminmax/lib/main.js:
```diff
@@ -0,0 +1,87 @@
+	var nanminmax;
+	if ( arguments.length === 0 ) {
+		nanminmax = incrminmax();
+	}
+	else {
```
```suggestion
	} else {
```

kgryte commented on lib/node_modules/@stdlib/stats/incr/nanminmax/lib/main.js:
```diff
@@ -0,0 +1,87 @@
+* // returns [ -5.0, 5.0 ]
+*/
+function incrnanminmax( out ) {
+	var nanminmax;
+	if ( arguments.length === 0 ) {
+		nanminmax = incrminmax();
```
```suggestion
	var minmax;
	if ( arguments.length === 0 ) {
		minmax = incrminmax();
```

kgryte commented on lib/node_modules/@stdlib/stats/incr/nanminmax/lib/main.js:
```diff
@@ -0,0 +1,87 @@
+		nanminmax = incrminmax();
+	}
+	else {
+		nanminmax = incrminmax( out );
```
```suggestion
		minmax = incrminmax( out );
```

kgryte commented on lib/node_modules/@stdlib/stats/incr/nanminmax/lib/main.js:
```diff
@@ -0,0 +1,87 @@
+	*/
+	function accumulator( x ) {
+		if ( arguments.length === 0 || isnan( x ) ) {
+			return nanminmax();
+		}
+		return nanminmax( x );
```
```suggestion
			return minmax();
		}
		return minmax( x );
```

kgryte commented on lib/node_modules/@stdlib/stats/incr/nanminmax/README.md:
```diff
@@ -0,0 +1,157 @@
+
+By default, the returned accumulator `function` returns the minimum and maximum as a two-element `array`. To avoid unnecessary memory allocation, the function supports providing an output (destination) object.
+
+
```
```suggestion
```

kgryte commented on lib/node_modules/@stdlib/stats/incr/nanminmax/README.md:
```diff
@@ -0,0 +1,157 @@
+
+#### incrnanminmax( \[out] )
+
+Returns an accumulator `function` which incrementally computes a minimum and maximum, ignoring `NaN` values.
```
```suggestion
Returns an accumulator function which incrementally computes a minimum and maximum, ignoring `NaN` values.
```

kgryte commented on lib/node_modules/@stdlib/stats/incr/nanminmax/README.md:
````diff
@@ -0,0 +1,157 @@
+var accumulator = incrnanminmax();
+```
+
+By default, the returned accumulator `function` returns the minimum and maximum as a two-element `array`. To avoid unnecessary memory allocation, the function supports providing an output (destination) object.
````
```suggestion
By default, the returned accumulator function returns the minimum and maximum as a two-element array. To avoid unnecessary memory allocation, the function supports providing an output (destination) object.
```

abhishekblue replied:
Removed this line as per previous comment (https://github.com/stdlib-js/stdlib/pull/5995#discussion_r1995031439)

kgryte commented on lib/node_modules/@stdlib/stats/incr/nanminmax/README.md:
```diff
@@ -0,0 +1,157 @@
+        v = NaN;
+    } else {
+        v = randu() * 100.0;
+    }
```
You're not passing any values to the accumulator here. Why?

abhishekblue committed:
made suggested change

Planeshifter committed:
chore: minor clean-up

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

Planeshifter commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into feat/incr-nanminmax

Planeshifter committed:
docs: remove extra comma

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>
